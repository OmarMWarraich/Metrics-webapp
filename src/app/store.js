import { configureStore, combineReducers } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import homeReducer from '../features/home/homeSlice';

const rootReducer = combineReducers({
  counter: counterReducer,
  home: homeReducer,
});

/* eslint-disable-next-line import/prefer-default-export */
export const store = configureStore({
  reducer: rootReducer,
});
