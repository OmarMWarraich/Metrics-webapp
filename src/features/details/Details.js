import React from 'react';
import { useSelector } from 'react-redux';
import { Card } from 'react-bootstrap';
import styles from './Details.module.css';
import PieToBarChart from '../home/Pie2BarChart';

const Details = () => {
  const Id = localStorage.getItem('id');
  const arr = useSelector((state) => state.home);
  const details = arr.filter((item) => item.CoinInfo.Id === Id);

  return (
    <>
      <div className={styles.container}>
        {details.map((item) => (
          <div key={item.CoinInfo.Id}>
            <Card
              className={styles.card}
            >
              <Card.Body
                className={styles.cardBody}
                style={{ border: '2px solid #002147' }}
              >
                <div className={styles.img_container}>
                  <PieToBarChart />
                </div>
                <div
                  className={styles.cardDetails}
                >
                  <div>
                    <Card.Title className={styles.title}>{item.CoinInfo.Name}</Card.Title>
                  </div>
                  <div>
                    <div>
                      <Card.Text className={styles.text}>
                        {item.CoinInfo.Id}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.text}>
                        {item.CoinInfo.FullName}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.text}>
                        {item.CoinInfo.Price}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.text}>
                        $(B)
                        {' '}
                        {(item.RAW.USD.MKTCAP / 1000000000).toFixed(2)}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>
            </Card>
            <Card
              key={item.CoinInfo.Id}
              id={item.CoinInfo.Id}
              className={styles.card2}
            >
              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Algorithm:</strong>
                        {item.CoinInfo.Algorithm}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>ProofType:</strong>
                        {item.CoinInfo.ProofType}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Rating:</strong>
                        {item.CoinInfo.Rating.Weiss.Rating}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>TAR:</strong>
                        {item.CoinInfo.Rating.Weiss.TechnologyAdoptionRating}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>

              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Block Time:</strong>
                        {item.CoinInfo.BlockTime}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Block Reward:</strong>
                        {item.CoinInfo.BlockReward}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Asset Launch Date:</strong>
                        {item.CoinInfo.AssetLaunchDate}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Max Supply:</strong>
                        {item.CoinInfo.MaxSupply}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>

              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>From Symbol:</strong>
                        {item.DISPLAY.USD.FROMSYMBOL}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>To Symbol:</strong>
                        {item.DISPLAY.USD.TOSYMBOL}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Market:</strong>
                        {item.DISPLAY.USD.MARKET}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Price:</strong>
                        {item.DISPLAY.USD.PRICE}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>

              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>LAST Volume:</strong>
                        {item.DISPLAY.USD.LASTVOLUME}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Volume 24 hour</strong>
                        {item.DISPLAY.USD.VOLUME24HOUR}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Volume 24 hour to:</strong>
                        {item.DISPLAY.USD.VOLUME24HOURTO}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Volume 24 hour from:</strong>
                        {item.DISPLAY.USD.OPEN24HOUR}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>

              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>High 24 hour:</strong>
                        {item.DISPLAY.USD.HIGH24HOUR}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Low 24 hour:</strong>
                        {item.DISPLAY.USD.LOW24HOUR}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Last Market:</strong>
                        {item.DISPLAY.USD.LASTMARKET}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Change 24 hour:</strong>
                        {item.DISPLAY.USD.CHANGE24HOUR}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>

              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Change Percent 24 hour:</strong>
                        {item.DISPLAY.USD.CHANGEPCT24HOUR}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Change 24 hour:</strong>
                        {item.DISPLAY.USD.CHANGE24HOUR}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Change hour %:</strong>
                        {item.DISPLAY.USD.CHANGEPCTHOUR}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Change hour:</strong>
                        {item.DISPLAY.USD.CHANGEHOUR}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>

              <Card.Body className={styles.cardLBody}>
                <div className={styles.cardLDetails}>
                  <div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Supply:</strong>
                        {item.DISPLAY.USD.SUPPLY}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Mkt Cap:</strong>
                        {item.DISPLAY.USD.MKTCAP}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Total Volume 24 hour:</strong>
                        {item.DISPLAY.USD.TOTALVOLUME24H}
                      </Card.Text>
                    </div>
                    <div>
                      <Card.Text className={styles.LText}>
                        <strong>Total Volume 24 hour to:</strong>
                        {item.DISPLAY.USD.TOTALVOLUME24HTO}
                      </Card.Text>
                    </div>
                  </div>
                </div>
              </Card.Body>
            </Card>
          </div>
        ))}
      </div>
    </>
  );
};

export default Details;
