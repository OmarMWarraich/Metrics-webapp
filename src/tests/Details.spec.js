import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import Details from '../features/details/Details';

it('Details page renders', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Details />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
