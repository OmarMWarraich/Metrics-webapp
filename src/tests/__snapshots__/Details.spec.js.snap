// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page renders 1`] = `
<div
  className="container"
/>
`;
